decidim-comments/app/frontend/comments/comments.component.jsx

```diff
@@ -0,0 +1,130 @@
+import { Component, PropTypes } from 'react';
+import { graphql }              from 'react-apollo';
+import gql                      from 'graphql-tag';
+import { filter }               from 'graphql-anywhere';
+import { I18n }                 from 'react-i18nify';
+
+import Application              from '../application/application.component';
+
+import CommentThread            from './comment_thread.component';
+import AddCommentForm           from './add_comment_form.component';
+
+import commentsQuery            from './comments.query.graphql';
+
+/**
+ * The core class of the Decidim Comments engine.
+ * It renders a collection of comments given a commentable id and type.
+ * @global
+ */
+export class Comments extends Component {
+  render() {
+    const { comments } = this.props;
+
+    return (
+      <div className="columns large-9" id="comments">
+        <section className="comments">
+          <div className="row collapse order-by">
+            <h2 className="order-by__text section-heading">
+              { I18n.t("components.comments.title", { count: comments.length }) }
+            </h2>
+          </div>
+          {this._renderCommentThreads()}
+          {this._renderAddCommentForm()}
+        </section>
+      </div>
+    );
+  }
+ 
+  /**
+   * Iterates the comment's collection and render a CommentThread for each one
+   * @private
+   * @returns {ReactComponent[]} - A collection of CommentThread components
+   */
+  _renderCommentThreads() {
+    const { comments } = this.props;
+
+    return comments.map((comment) => (
+      <CommentThread 
+        key={comment.id} 
+        comment={filter(CommentThread.fragments.comment, comment)} 
+      />
+    ))
+  }
+ 
+  /**
+   * If session's current user is present it renders the add comment form
+   * @private
+   * @returns {Void|ReactComponent} - A AddCommentForm component or nothing
+   */
+  _renderAddCommentForm() {
+    const { session, commentableId, commentableType } = this.props;
+    
+    if (session.currentUser) {
```

Comment:
Are we sure about this, shouldn't we have the concept of abilities at the frontend?